frontends/web/src/components/alert/Alert.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2018 Shift Devices AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, h } from 'preact';
+import { translate } from 'react-i18next';
+import Dialog from '../dialog/dialog';
+import { Button } from '../forms';
+
+let alertUser: (message: string, callback?: () => void) => void;
+
+interface Props {
+    t: () => void;
+}
+
+interface State {
+    active: boolean;
+    message?: string;
+}
+
+@translate()
+class Alert extends Component<Props, State> {
+    private button!: Component;
```

Comment:
Ok, I see what you are doing here: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#strict-class-initialization. I'd be very careful with using "definite assignment assertion" – especially since the field is not actually initialized after the constructor. It requires that you reasoned through all assignments and usages of the variables. Please add a comment why your assumption (that `button` is only used initialized) holds (like `// Initialized after initial rendering.`).